### DISCUSSION TITLE
Editor warning for non-Node Autoload scripts

### DISCUSSION THREAD

**Discussion by archiebaldry** [godotengine/godot-proposals#8720] [Scripting]
I recently ran into an issue where my Autoload script wasn't working as expected. This was because I had made the script extend from the `Object` class and not the `Node` class (or a derivative of `Node`). I think the editor should give some kind of warning when trying to use a non-Node script as an Autoload. [Here's a relevant Reddit post on r/godot](https://www.reddit.com/r/godot/comments/akm2kq/should_all_singletons_extends_for_node/). Please forgive me if I have misunderstood how Autoload works!

**Comment by archiebaldry** [comment]
This suggestion may be better suited in the Editor category, but I'll leave it here for now. Please feel free to move it if required! :)

**Comment by Mickeon** [comment]
This may be intrinsically solved by https://github.com/godotengine/godot/pull/81886 and been already proposed in https://github.com/godotengine/godot-proposals/issues/7710 , because all scene nodes inherit from Node.
